src/pykeen/datasets/__init__.py

```diff
@@ -184,4 +158,4 @@ def get_dataset(
 
 def has_dataset(key: str) -> bool:
     """Return if the dataset is registered in PyKEEN."""
-    return normalize_string(key) in datasets
+    return dataset_resolver.lookup(key) is not None
```

Comment:
This could be added as a functionality to `class_resolver` :slightly_smiling_face: